kafka-client/src/main/java/dev/responsive/kafka/api/async/internals/AsyncThreadPool.java

```diff
@@ -192,7 +192,7 @@ public <KOut, VOut> void scheduleForProcessing(
           .whenComplete((r, t) -> {
             inFlightForTask.remove(event);
           })
-          .handle((processingException, fatalException) -> {
+          .whenComplete((processingException, fatalException) -> {
```

Comment:
this is the main fix, as explained in the commit msg

Reply:
Good catch, this was my mistake -- I thought `handle` just let you change the return value but didn't realize it would wipe out the exception as well.

FYI you can probably remove the comment on line 206 (or move it up here if you think it has value, I went overboard with the comments I know)